insights/core/__init__.py

```diff
@@ -765,9 +765,29 @@ class YAMLParser(Parser, LegacyItemAccess):
     A parser class that reads YAML files.  Base your own parser on this.
     """
 
+    ignore_lines = []
+    """
+    The first keywords of the lines that need to ignored in the parsing process.
+    * All keywords added in this list must be in lowercase.
+    """
+
     def parse_content(self, content):
+        def _ignore_lines(content, ignore_lines):
+            idxs = []
```

Comment:
From the usage of the `idxs`, what about using it as `idxs = set()`, to help speed up a little bit in the item adding and `in` checking below?

Reply:
For both "item adding" and "x in s" operations, it takes O(1) time complexity (on average case) for `set()` type , and takes O(n) for `list()` type.  

As we do "item adding" inside  `_ignore_lines()`, and do the `in` checking at `for i, l in enumerate(content) if i not in idxs]`, both can benefit from the `set()` type than `list()`, just, benefit a little bit as we don't likely have a long `idxs` typically.

Reply:
Hm... I cannot agree with you about this. As `set` stores items with a hash algorithm, no matter long or short, it cost more than a `list` when storing one item. The only good point I can think for a `set` is when doing a `search` which is not used in this case.

Reply:
No worries, I removed the pre-searching of the `indexes`, change to skip the lines directly which will save one pass of `content` iteration.